src/inference/src/os/lin/lin_system_conf.cpp

```diff
@@ -43,13 +43,85 @@ CPU::CPU() {
         return 0;
     };
 
+    auto CheckValidCpu = [&]() {
+        cpu_set_t mask;
+        CPU_ZERO(&mask);
+
+        if (sched_getaffinity(0, sizeof(cpu_set_t), &mask) == -1) {
+            return -1;
+        }
+
+        int total_proc = 0;
+        std::vector<int> cores_list;
+        std::vector<int> phy_core_list;
+        std::vector<std::vector<int>> valid_cpu_mapping_table;
+
+        for (int i = 0; i < _processors; i++) {
+            if (CPU_ISSET(i, &mask)) {
+                total_proc++;
+                cores_list.emplace_back(_cpu_mapping_table[i][CPU_MAP_CORE_ID]);
+                valid_cpu_mapping_table.emplace_back(_cpu_mapping_table[i]);
+                if (_cpu_mapping_table[i][CPU_MAP_CORE_TYPE] == MAIN_CORE_PROC) {
+                    phy_core_list.emplace_back(_cpu_mapping_table[i][CPU_MAP_GROUP_ID]);
+                }
+            }
+        }
+
+        if (total_proc == _processors) {
+            return 0;
+        } else {
+            _processors = total_proc;
+            _cpu_mapping_table.swap(valid_cpu_mapping_table);
+            for (auto& row : _proc_type_table) {
+                std::fill(row.begin(), row.end(), 0);
+            }
+            for (auto& row : _cpu_mapping_table) {
+                if (row[CPU_MAP_CORE_TYPE] == HYPER_THREADING_PROC) {
+                    auto iter = std::find(phy_core_list.begin(), phy_core_list.end(), row[CPU_MAP_GROUP_ID]);
+                    if (iter == phy_core_list.end()) {
+                        row[CPU_MAP_CORE_TYPE] = MAIN_CORE_PROC;
+                    }
+                }
+                _proc_type_table[0][ALL_PROC]++;
+                _proc_type_table[0][row[CPU_MAP_CORE_TYPE]]++;
+                if (_proc_type_table.size() > 1) {
+                    _proc_type_table[row[CPU_MAP_SOCKET_ID] + 1][ALL_PROC]++;
+                    _proc_type_table[row[CPU_MAP_SOCKET_ID] + 1][row[CPU_MAP_CORE_TYPE]]++;
+                }
+            }
+
+            if (_proc_type_table.size() > 1) {
+                long unsigned int n = 0;
+
+                while (n < _proc_type_table.size()) {
+                    if (0 == _proc_type_table[n][ALL_PROC]) {
+                        _proc_type_table.erase(_proc_type_table.begin() + n);
+                        n--;
```

Comment:
Could it guarantee `n>0` and then `n--`? Maybe some strange platform may hit it, so we need protect it from this case.

Reply:
updated